plugins/osdn/podinfoserver.go

```diff
@@ -0,0 +1,138 @@
+package osdn
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"path"
+	"syscall"
+
+	"github.com/gorilla/mux"
+
+	"github.com/openshift/openshift-sdn/pkg/netutils"
+	"github.com/openshift/openshift-sdn/plugins/osdn/api"
+)
+
+type PodInfoServer struct {
+	http.Server
+
+	registry *Registry
+	vnidMap  *VNIDMap
+	nodeName string
+
+	clusterNetwork string
+	nodeNetwork    string
+	nodeGateway    string
+	mtu            uint
+}
+
+func NewPodInfoServer(r *Registry, v *VNIDMap, nodeName string) *PodInfoServer {
+	router := mux.NewRouter()
+ 
+	s := &PodInfoServer{
+		registry: r,
+		vnidMap:  v,
+		nodeName: nodeName,
+		Server:   http.Server{
+			Handler: router,
+		},
+	}
+
+	router.NotFoundHandler = http.HandlerFunc(http.NotFound)
+	router.Methods("GET").Path("/network").HandlerFunc(s.getNetwork)
+	router.Methods("GET").Path("/pods/{namespace}/{name}").HandlerFunc(s.getPod)
+
+	return s
+}
+
+func (s *PodInfoServer) Start(clusterNetworkCIDR, localSubnetCIDR string, mtu uint) error {
+	s.clusterNetwork = clusterNetworkCIDR
+	s.nodeNetwork = localSubnetCIDR
+	_, ipnet, err := net.ParseCIDR(localSubnetCIDR)
+	s.nodeGateway = netutils.GenerateDefaultGateway(ipnet).String()
+	s.mtu = mtu
+
+	if err := syscall.Unlink(api.PodInfoSocketPath); err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("failed to removing old pod info socket: %v", err)
+	}
+
+	dirName := path.Dir(api.PodInfoSocketPath)
+	if err := os.MkdirAll(dirName, 0700); err != nil {
+		return fmt.Errorf("failed to create pod info socket directory: %v", err)
+	}
+
+	mask := syscall.Umask(0777)
+	l, err := net.Listen("unix", api.PodInfoSocketPath)
+	syscall.Umask(mask)
```

Comment:
I meant that you might be changing the umask for an unrelated operation happening at the same time in another thread.